### ISSUE TITLE
Integration tests not enhancing domain classes in plugin projects

### ISSUE THREAD

**Issue by pledbrook** [grails/grails-core#9432]
I was playing around with the Hibernate Filter plugin after someone said that the tests weren't working. It seems that in a plugin project, integration tests don't enhance the domain classes and so you end up with  method missing exceptions.

You can try it yourself by grabbing the Hibernate Filter source and running the integration tests. You'll first have to remove the `@Ignore` annotation added in [this commit](https://github.com/fingo/grails-hibernate-filter/commit/cdc6b04da5ee9a7e1f25782186793c80f1945e65), which will also link you the source repo. I don't mind myself whether this feature is available or not, but if it's not supposed to work, then there should be an error message that integration tests should not be used in plugin projects and test application projects should be used instead as part of a multi-project build.

**Comment by graemerocher** [comment]
As far as I can tell the plugin doesn't declare a dependency on the `hibernate` plugin so this seems to be expected, or am I missing something?

**Comment by pledbrook** [comment]
I modified the dependencies block so it has

    compile "org.grails:grails-datastore-gorm-hibernate:4.0.4.RELEASE"
    compile "org.grails.plugins:hibernate", {
        exclude module: "grails-datastore-gorm-hibernate4"
    }

I also added

    classpath "org.grails.plugins:hibernate:4.3.10.5"

to the `buildscript` dependencies. I'm still seeing a `MissingMethodException` for `Foo.addToBars()` (I modified the `setup()` method slightly, renamed it to `setupData()` and call it explicitly from the `given` block of each feature method).

Is the plugin the bit that enhances the domain classes then? What does adding the _grails-datastore-gorm-hibernate_ dependency alone do for you? That dependency was there from the start, so I'm just wondering whether it served any purpose other than to provide certain required classes.
